### ISSUE TITLE
Access to real require or real import

### ISSUE THREAD

**Issue by Yuri6037** [webpack/webpack#11455]
Hello,

I can't post on StackOverflow so I ask that here. Is there any way to bypass webpack version of require and use the real require or the real async import that just import js files from random locations?
I ask that for the purpose of module based electron application: it is similar to VSCode extension system that can run arbitary JS files coming from extension where those JS files would not forcely be built with the application.

Currently using require with an absolute path just results in file not found. Using async import with absolute path does work however throws a warning on console telling that import was used with an expression.

**Comment by bebraw** [comment]
I think you might be looking for `__non_webpack_require__`. See https://github.com/webpack/webpack/issues/4175#issuecomment-277232067.

**Comment by evilebottnawi** [comment]
Yes, answer here https://github.com/webpack/webpack/issues/4175#issuecomment-277232067, feel free to feedback, we need docs this

**Comment by Yuri6037** [comment]
```TypeScript
// @ts-ignore
const obj = __non_webpack_require__("/home/yuri/Projects/ERPMod/Client/build.js");
```

Produces:

```
Uncaught ReferenceError: require is not defined
    at Object../src/App.tsx (App.tsx:12)
    at __webpack_require__ (bootstrap:19)
    at bootstrap:83
    at bootstrap:83
```

**Comment by evilebottnawi** [comment]
Because `require` doesn't exists in browser

**Comment by Yuri6037** [comment]
Ok, then my best way is to rely on trying to get contextBridge to not crash miserably and expose some wrapped fs.readFileSync and eval on the client side

**Comment by Yuri6037** [comment]
Well actually not: it seam that electron has done all they could so you cannot create a dynamic module system!

**Comment by evilebottnawi** [comment]
@Yuri6037 I think you look this https://webpack.js.org/concepts/module-federation/

**Comment by evilebottnawi** [comment]
Alternative you can use `import(/* webpackIgnore */ '/dist/file.js').then()`

**Comment by Yuri6037** [comment]
Thank you but /* webpackIgnore */ has absolutely no effect.
Also adding .then made the client throw an exception:

```
Uncaught (in promise) Error: Cannot find module '/home/yuri/Projects/ERPMod/Client/build.js'
    at webpackEmptyContext (src sync:2)
    at App.tsx:35
```

**Comment by evilebottnawi** [comment]
I can't help because you don't provide what you want achieve and what target/configuration your use

**Comment by Yuri6037** [comment]
I need a way to inject arbitary JS files that exists on the local file system into the rendering context of Electron. Like IMvcBuilder.AddApplicationPart in Asp.Net Core

**Comment by Yuri6037** [comment]
I think I just found something: when a script block is injected by force into the index.html from the main context, the rendering context seam to not complain and just run my evil JS injection test.

I have a question to Webpack authors: is it guarenteed that a single .js file will be generated from n number of JS files when referenced as an output target in webpack config? Is it guarenteed that the name of the output JS file will always be what the config file says and not some random hash?

**Comment by evilebottnawi** [comment]
@Yuri6037 

> I need a way to inject arbitary JS files that exists on the local file system into the rendering context of Electron. Like IMvcBuilder.AddApplicationPart in Asp.Net Core

Ask this in `electron` repo

> I have a question to Webpack authors: is it guarenteed that a single .js file will be generated from n number of JS files when referenced as an output target in webpack config?

Based on your configuration

> Is it guarenteed that the name of the output JS file will always be what the config file says and not some random hash?

Yes

Please use QA service for questions, it is bug tracker, you ask help but doesn't provide versions/configurations/structure/goal/etc

**Comment by Yuri6037** [comment]
Thanks a lot.
So if I do

```
        entry: "./src/App.tsx"
        output: {
            path: process.cwd() + "/dist/release",
            filename: "App.js"
        }
```

Then one file will be generated named App.js in dist/release and it will always be named as such and will always be one, right?

**Comment by evilebottnawi** [comment]
Yes
